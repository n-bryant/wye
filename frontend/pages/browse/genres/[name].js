import React from "react";
import PropTypes from "prop-types";

import { useRouter } from "next/router";

import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../../lib/classNamePrefix";
import { withStyles } from "@material-ui/core/styles";

import GamesFilterWidget from "../../../src/components/Browse/GamesFilterWidget";
import styles from "./[name].styles";

/**
 * renders a GamesFilterWidget, filtering for games of the provided genre
 */
export const GenrePage = props => {
  const classnames = GenrePage.classnames(props);
  const router = useRouter();

  const genre = router.query.name;
  const subtitle = `Browse ${genre} titles`;
  const config = {
    filters: {
      gameFilters: {
        genres_in: [genre]
      }
    },
    orderBy: ["USER_RATING"],
    sortOrder: "DESC"
  };

  return (
    <div className={classnames.root()}>
      <GamesFilterWidget
        title={genre}
        subtitle={subtitle}
        initialFilters={config}
      />
    </div>
  );
};
GenrePage.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}GenrePage`
);
GenrePage.propTypes = {
  // styles to apply
  classes: PropTypes.shape({
    root: PropTypes.string
  })
};
GenrePage.defaultProps = {
  classes: {}
};

// apply styles
export const StyledGenrePage = withStyles(styles)(GenrePage);

export default StyledGenrePage;
