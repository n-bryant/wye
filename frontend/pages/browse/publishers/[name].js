import React from "react";
import PropTypes from "prop-types";

import { useRouter } from "next/router";

import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../../lib/classNamePrefix";
import { withStyles } from "@material-ui/core/styles";

import GamesFilterWidget from "../../../src/components/Browse/GamesFilterWidget";
import styles from "./[name].styles";

/**
 * renders a GamesFilterWidget, filtering for games of the provided publisher
 */
export const PublisherPage = props => {
  const classnames = PublisherPage.classnames(props);
  const router = useRouter();

  const publisher = router.query.name;
  const subtitle = `Browse ${publisher} titles`;
  const config = {
    filters: {
      gameFilters: {
        publishers_in: [publisher]
      }
    },
    orderBy: ["USER_RATING"],
    sortOrder: "DESC"
  };

  return (
    <div className={classnames.root()}>
      <GamesFilterWidget
        title={publisher}
        subtitle={subtitle}
        initialFilters={config}
      />
    </div>
  );
};
PublisherPage.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}PublisherPage`
);
PublisherPage.propTypes = {
  // styles to apply
  classes: PropTypes.shape({
    root: PropTypes.string
  })
};
PublisherPage.defaultProps = {
  classes: {}
};

// apply styles
export const StyledPublisherPage = withStyles(styles)(PublisherPage);

export default StyledPublisherPage;
