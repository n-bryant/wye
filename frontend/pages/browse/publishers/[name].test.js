import { PublisherPage } from "./[name]";

jest.mock("next/router", () => ({
  useRouter: () => ({
    query: {
      name: "name"
    }
  })
}));

describe("PublisherPage", () => {
  const props = {
    calsses: {
      root: "root"
    }
  };

  it("should render successfully", () => {
    const wrapper = shallow(<PublisherPage {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
