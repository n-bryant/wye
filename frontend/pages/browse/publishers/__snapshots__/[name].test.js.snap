// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublisherPage should render successfully 1`] = `
<div
  className="wye-PublisherPage"
>
  <WithWidth(WithStyles(GamesFilterWidget))
    initialFilters={
      Object {
        "filters": Object {
          "gameFilters": Object {
            "publishers_in": Array [
              "name",
            ],
          },
        },
        "orderBy": Array [
          "USER_RATING",
        ],
        "sortOrder": "DESC",
      }
    }
    subtitle="Browse name titles"
    title="name"
  />
</div>
`;
